### PR TITLE
Add migration to clean up old verification state

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Clean up the redux store after the verification changes from this PR: https://github.com/celo-org/wallet/pull/377/

### Other changes

N/A

### Tested

Manually and with a unit test

### How others should test

Upgrade for an older version (1.14.x or older) to the latest one and make sure that the verification status is not lost and there are no other unexpected things related to verification.

### Related issues

- Fixes #116

### Backwards compatibility

N/A

### CODE REVIEW

**Review by jeanregisser — COMMENTED** [review]
Thanks! 👍 

We also need the following:
- bump version in the redux config
- add a new version in `test/schema.ts` so the testing schema is correct

**Comment by gnardini** [comment]
Did both those things but I don't really understand why we have different versions in `test/schemas.ts`. Wouldn't it be easier to read if we had a full schema with all the properties instead of having it be divided by version? This version doesn't match the migration version either, so I feel like I'm missing something.

**Comment by jeanregisser** [comment]
Would be easier to read, but the reason why it's divided by version is so we can use them to test migrations.
I think we should go back to have the test schemas version in sync as a good practice:
https://github.com/celo-org/wallet/blob/0db5533c13f4b8e787497c4333cb805fa3711e40/packages/mobile/test/schemas.ts#L368-L369

File: packages/mobile/test/schemas.ts [hunk 1]
```diff
@@ -592,6 +592,22 @@ export const v9Schema = {
...
+      'attestationCodes',
+      'acceptedAttestationCodes',
+      'attestationInputStatus',
+      'numCompleteAttestations',
+      'verificationStatus',
+      'hasSeenVerificationNux',
+      'lastRevealAttempt'
+    ),
+  },
+}
```
Comment (jeanregisser) [inline packages/mobile/test/schemas.ts:609]: `verify` key should be moved here from https://github.com/celo-org/wallet/pull/377/files#diff-a0850796be5a07a6363631e739d56ea69fa51c26852655610c6f7650285b9a2e

Comment (jeanregisser) [reply packages/mobile/test/schemas.ts:609]: and `identity` keys shouldn't have been removed from `v9Schema` in https://github.com/celo-org/wallet/pull/377/files#diff-a0850796be5a07a6363631e739d56ea69fa51c26852655610c6f7650285b9a2e

**Comment by gnardini** [comment]
All right, I haven't been using the schemas from that file for migration tests but I can see why we would want that. In that case, I renamed the latest one to `v13Schema`, added a `storeVersion` field to it and a test that checks that it's in sync with the store. Hopefully this helps us to keep them in sync :) someone could still technically set the `storeVersion` field of the `v13Schema` to 14, but that should be flagged in a review, otherwise we could move the test schema file to an object similar to the migrations one.

**Review by jeanregisser — APPROVED** [review]

**Comment by Celoqa** [comment]
hi @gnardini we have verified this issue on latest Android internal build V1.15.0 & Testflight build V1.15.0(63) and observe the followings:

- [ ] User is able to see completed verification flow when user upgraded to V1.14-V1.15 if user already verified phone number in V1.14

- [ ] User is able to see all data correctly cUSD/CELO

- [ ] First & Second code fields are shown blank when user ungraded to V1.14-V1.15 given below are the steps to repro issue:

  - Install V1.14 build 
  - Paste only first & second fields Keep third code as blank then skip the verification
  - Make sure user should able to see first & second codes when resume on V1.14 build
  - Upgrade it to V1.15 & go to verification flow
  - Observe First & second fields are shown blank.
[Attachment](https://drive.google.com/file/d/1EI8fygqbfTj4jNEb_GX5SLiDhny5qZ4r/view?usp=sharing)
Please let us know we need to open new issue for this.
Thanks!

**Comment by gnardini** [comment]
The described behavior is fine, thanks for checking!

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R32](https://valora.testquality.com/project/12128/testcase/286103/requirement/11365). Do not remove or edit.

**Pull request merged**